### PR TITLE
[ENG-3533] File detail revisions

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-3533]
-   Feature flag: n/a

## Purpose
- Add file revision content to the file-detail page

## Summary of Changes
- Rewrite `file-version` component to match this new implementation
- Use newly revamped `file-revision` component in guid-file route
- Allow MFR to accept different versions of the file
- TODO: Rewrite `file-version` test and update file-detail acceptance test

## Screenshot(s)
Desktop:
image

Mobile:
image

## Side Effects
- None

## QA Notes
- TODO

[ENG-3533]: https://openscience.atlassian.net/browse/ENG-3533?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/guid-file/controller.ts [hunk 1]
```diff
@@ -24,6 +32,29 @@ export default class GuidFile extends Controller {
...
 
+    // get versions from Waterbutler
+    @task
+    @waitFor
+    async loadRevisions() {
+        let response;
+        try {
+            response = await $.ajax({
+                type: 'GET',
+                url: `${this.model.links.download}?revisions=&`,
```
Comment (brianjgeiger) [inline app/guid-file/controller.ts:43]: So this is provided by WB, and doesn't paginate? Interesting.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Overall, looking great. I would say let's get away from using jQuery whenever possible. Also, could we add in a signifier by the filename to indicate when you're viewing a version and which version you're viewing? Something like `Filename.jpg (Version 14)`.

File: app/guid-file/controller.ts (hunk 1 above)
Comment (futa-ikeda) [reply app/guid-file/controller.ts:43]: Yea, I tried a file with 26 revisions, and it seems that it doesn't paginate (at least at 26). I'm sure there are projects out there with files that have undergone many more revisions than that, so I may snoop/ask if there is indeed a point in which we need to worry about pagination

File: app/packages/files/file.ts [hunk 2]
```diff
@@ -65,6 +89,13 @@ export default abstract class File {
         return [];
     }
 
+    async getRevisions() {
```
Comment (brianjgeiger) [inline app/packages/files/file.ts:92]: Should this be a task? We shouldn't need to restart it, but we probably want to show the pulser while we're loading information.

File: app/packages/files/file.ts [hunk 3]
```diff
@@ -81,4 +107,13 @@ export default abstract class File {
     async delete() {
         await this.fileModel.delete();
     }
+
+    @task
+    @waitFor
+    async getRevisions() {
+        const responseObject = await fetch(`${this.links.download}?revisions=&`);
```
Comment (adlius) [inline app/packages/files/file.ts:114]: This URL doesn't seem to be getting the revisions.

**Review by adlius — CHANGES_REQUESTED** [review]

**Pull request merged**